0001-typedb/02-dev/06-infer.md

```diff
@@ -0,0 +1,336 @@
+                    ex = 0
+                    for explanation in explain_iterator:
+                        ex += 1
+                        
+                        print("\nRead result #:", i, ", File path:", item.get("o-path").as_attribute().get_value())
+                        print("Explainable #:", e, ", Explained variable:", explainable)
+                        print("Explainable object:", explainable_relations[explainable])
+                        print("Explainable part of query:", explainable_relations[explainable].conjunction())
+                        print("Explanation #:", ex)
+                        
+                        print("\nRule: ", explanation.rule().get_label())
+                        print("Condition: ", explanation.condition())
+                        print("Conclusion: ", explanation.conclusion())
+                        print("Variables used in explanation: ", explanation.variable_mapping())
+                        print("----------------------------------------------------------")
+```
+
+The script above runs the query from the [example](#example) in the previous section. The inference option provides 
+the result of 10 files (by default in the IAM database from the [Quickstart guide](../01-start/03-quickstart.md)). 
+And `explain` option enables the `explainables` to be received and used in the **explain** queries (one explain query 
+for each result that needs to be explained).
+
+#### Output
+
+The result should be similar to the following:
+
+<!-- test-ignore -->
+```
+Read result #: 10 , File path: zlckt.ts
+Explainable #: 1 , Explained variable: pe
+Explainable object: &lt;typedb.concept.answer.concept_map._ConceptMap.Explainable object at 0x105cb34f0>
+Explainable part of query: { $pe (permitted-subject:$p, permitted-access:$ac); $pe isa permission; }
+Explanation #: 1
+
+Rule:  add-view-access
+
+Condition:  [_1/_StringAttribute[action-name:0x836f800328000b6d6f646966795f66696c65]][_2/_StringAttribute[action-name:0x836f8003280009766965775f66696c65]][_3/_Relation[permission:0x847080038000000000000001]][ac_modify/_Relation[access:0x8470800a8000000000000003]][ac_view/_Relation[access:0x8470800a8000000000000011]][modify/_Entity[operation:0x826e800c8000000000000001]][obj/_Entity[file:0x826e80098000000000000004]][subj/_Entity[person:0x826e80018000000000000001]][view/_Entity[operation:0x826e800c8000000000000000]]
+
+Conclusion:  [_/_Relation[permission:0x847080037fffffffffffffff]][_permission/_RelationType[label: permission]][_permission:permitted-access/_RoleType[label: permission:permitted-access]][_permission:permitted-subject/_RoleType[label: permission:permitted-subject]][ac_view/_Relation[access:0x8470800a8000000000000011]][subj/_Entity[person:0x826e80018000000000000001]]
+
+Variables used in explanation:  {'p': {'subj'}, 'ac': {'ac_view'}, 'pe': {'_'}}
+
+----------------------------------------------------------
+```
```

Comment:
We should fix the format of the output rather than including this in the docs.

Reply:
What are you talking about exactly? What needs fixing? 
If it's about the horizontal line it is required because we have multiple very similar blocks with a lot of hardly readable data. The data is provided in the same format it is returned by a method (deliberately, to familiarize users with what to expect).

Reply:
Sorry I wasn't clear in my comment.
I meant we (/the engineering team) should fix the output of the typedb clients for explanations.

Reply:
Well, we don't know when it will be fixed, how, and even whether it will be fixed at all. Hence I'm documenting it as it is now. If it'll change — let me know and I'll update the docs.